packages/gatsby/src/utils/page-data.js

```diff
@@ -35,7 +40,71 @@ const write = async ({ publicDir }, page, result) => {
   await fs.outputFile(filePath, bodyStr)
 }
 
+const getChangedPageDataKeys = (state, cachedPageData) => {
+  if (cachedPageData && state.pageData) {
+    const pageKeys = []
+    state.pageData.forEach((resultHash, key) => {
```

Comment:
```suggestion
    state.pageData.forEach((newPageDataHash, key) => {
```

and then we can also remove extra assignment on line 50